dbms/programs/client/Client.cpp

```diff
@@ -563,9 +563,18 @@ class Client : public Poco::Util::Application
         if (is_interactive)
         {
             std::cout << "Connected to " << server_name
-                      << " server version " << server_version
-                      << " revision " << server_revision
-                      << "." << std::endl << std::endl;
+                << " server version " << server_version
+                << " revision " << server_revision
+                << "." << std::endl << std::endl;
+
+            if (VERSION_MAJOR < server_version_major
```

Comment:
It isn't a lexicographical compare. It will fail for 19.1 vs 18.2.

`std::tie(a1, b1, c1) < std::tie(a2, b2, c2)`

Reply:
Yes, that was stupid. Fixed.